codegenerator/cli/templates/static/codegen/src/ink/components/CustomHooks.res

```diff
@@ -0,0 +1,94 @@
+open Belt
+module InitApi = {
+  type ecosystem = Evm | Fuel
+  type body = {
+    envioVersion: option<string>,
+    envioApiToken: option<string>,
+    ecosystem: ecosystem,
+    hyperSyncNetworks: array<int>,
+    rpcNetworks: array<int>,
+  }
+
+  let bodySchema = S.object(s => {
+    envioVersion: s.field("envioVersion", S.option(S.string)),
+    envioApiToken: s.field("envioApiToken", S.option(S.string)),
+    ecosystem: s.field("ecosystem", S.union([S.literal(Evm), S.literal(Fuel)])),
+    hyperSyncNetworks: s.field("hyperSyncNetworks", S.array(S.int)),
+    rpcNetworks: s.field("rpcNetworks", S.array(S.int)),
+  })
+
+  let makeBody = (~envioVersion, ~envioApiToken, ~config: Config.t) => {
+    let ecosystem = ref(Evm)
+    let hyperSyncNetworks = []
+    let rpcNetworks = []
+    config.chainMap
+    ->ChainMap.values
+    ->Array.forEach(({syncSource, chain}) => {
+      switch syncSource {
+      | HyperSync(_) => hyperSyncNetworks
+      | HyperFuel(_) =>
+        ecosystem := Fuel
+        hyperSyncNetworks
+      | Rpc(_) => rpcNetworks
+      }
+      ->Js.Array2.push(chain->ChainMap.Chain.toChainId)
+      ->ignore
+    })
+
+    {
+      envioVersion,
+      envioApiToken,
+      ecosystem: ecosystem.contents,
+      hyperSyncNetworks,
+      rpcNetworks,
+    }
+  }
+
+  type messageKind = | @as("warning") Warning | @as("destructive") Destructive | @as("info") Info
+  type message = {
+    kind: messageKind,
+    content: string,
+  }
+
+  let messageSchema = S.object(s => {
+    kind: s.field("kind", S.union([S.literal(Warning), S.literal(Destructive), S.literal(Info)])),
```

Comment:
```suggestion
    kind: s.field("kind", S.union([S.literal(Warning), S.literal(Destructive), S.literal(Info), S.string->S.variant(_ => General)])),
```

Reply:
Ah, well, we can handle it on the server, since we know the version. But I think it's still nice to have a white colored message.